### PR TITLE
Doxygen libs

### PR DESCRIPTION
**Pull request by nmathewson**

### CODE REVIEW

File: src/feature/hs/feature_hs.dox [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /**
 @dir /feature/hs
-@brief feature/hs
+@brief feature/hs: v3 (current) onion service protocol
```
Comment (dgoulet-tor) [inline src/feature/hs/feature_hs.dox]: I would be super greatful if you would put here a ref to the `rend-spec-v3.txt` :D like it was done above:

[dir-spec.txt](https://gitweb.torproject.org/torspec.git/tree/dir-spec.txt),

File: src/mainpage.dox [hunk 2]
```diff
@@ -3,9 +3,120 @@
...
+events.  As Tor runs over time, other events will open, and new events
+will be scheduled.
+
+The codebase is divided into a few top-level subdirectories, each of
+which contains several sub-modules.
+
+   - `ext` -- Code maintained elsewhere that we include in the Tor
+     source distribution.
+
+   - \refdir{lib} -- Lower-level utility code, not necessarily
```
Comment (dgoulet-tor) [inline src/mainpage.dox]: Not sure `refdir` outputs properly:

I get:

`./src/lib "lib" – Lower-level utility code, not necessarily tor-specific.`

File: src/mainpage.dox [hunk 3]
```diff
@@ -3,9 +3,120 @@
...
+started.  This means, however, that some of the older pieces of Tor will have
+some "code smell" in them that could stand a brisk refactoring.  So when we
+describe a piece of code, we'll sometimes give a note on how it got that way,
+and whether we still think that's a good idea.
+
+This document is not an overview of the Tor protocol.  For that, see the
+design paper and the specifications at https://spec.torproject.org/ .
+
+For more information about Tor's coding standards and some helpful
+development tools, see doc/HACKING in the Tor repository.
```
Comment (dgoulet-tor) [inline src/mainpage.dox]: Probably would be nice to have a ref to `doc/HACKING` so user can click on.

File: src/feature/rend/feature_rend.dox [hunk 4]
```diff
@@ -1,4 +1,5 @@
 /**
 @dir /feature/rend
-@brief feature/rend
+@brief feature/rend: version 2 (old) hidden services
```
Comment (dgoulet-tor) [inline src/feature/rend/feature_rend.dox]: Same goes for this with `rend-spec.txt`.

File: src/feature/dirauth/feature_dirauth.dox [hunk 5]
```diff
@@ -1,4 +1,6 @@
 /**
-@dir feature/dirauth
-@brief feature/dirauth
+@dir /feature/dirauth
+@brief feature/dirauth: Directory authority implementation.
+
+This module handles running Tor as a directory authority.
```
Comment (dgoulet-tor) [inline src/feature/dirauth/feature_dirauth.dox]: Maybe ref to `dir-spec.txt`.

File: src/mainpage.dox (hunk 2 above)
Comment (nmathewson) [reply src/mainpage.dox]: Hm. It works for me -- let's see how it works when we build the website.

**Pull request merged**